test/hashing.spec.js

```diff
@@ -0,0 +1,1347 @@
+              rawData: 'https://example.org/giai/401234599999',
+            },
+          },
+        ],
+      }, {});
+      expect(str).to.be.equal('sensorElementListsensorElementsensorMetadatatime=2019-04-02T13:05:00.000ZdeviceID=https://id.gs1.org/8004/4000001111deviceMetadata=https://id.gs1.org/8004/4000001111rawData=https://id.gs1.org/8004/401234599999');
+    });
+
+    it('should replace cbv vocabulary', () => {
+      const str = eventToPreHashedString({
+        bizStep: 'repairing',
+      }, {});
+      expect(str).to.be.equal('bizStep=https://ns.gs1.org/cbv/BizStep-repairing');
+      const str2 = eventToPreHashedString({
+        disposition: 'damaged',
+      }, {});
+      expect(str2).to.be.equal('disposition=https://ns.gs1.org/cbv/Disp-damaged');
+      const str3 = eventToPreHashedString({
+        bizTransactionList: [
+          {
+            type: 'desadv',
+            bizTransaction: 'urn:epcglobal:cbv:bt:5200001000008:4711',
+          },
+          {
+            type: 'inv',
+            bizTransaction: 'urn:epcglobal:cbv:bt:5200001000008:RE1099',
+          },
+        ],
+      }, {});
+      expect(str3).to.be.equal('bizTransactionListbizTransaction=urn:epcglobal:cbv:bt:5200001000008:4711type=https://ns.gs1.org/cbv/BTT-desadvbizTransaction=urn:epcglobal:cbv:bt:5200001000008:RE1099type=https://ns.gs1.org/cbv/BTT-inv');
+      const str4 = eventToPreHashedString({
+        destinationList: [
+          {
+            destination: 'urn:epc:id:sgln:4012345.00012.0',
+            type: 'location',
+          },
+          {
+            destination: 'urn:epc:id:pgln:4012345.00000',
+            type: 'owning_party',
+          },
+          {
+            type: 'possessing_party',
+            destination: 'urn:epc:id:pgln:4012345.00000',
+          },
+        ],
+      }, {});
+      expect(str4).to.be.equal('destinationListdestination=https://id.gs1.org/414/4012345000122type=https://ns.gs1.org/cbv/SDT-locationdestination=https://id.gs1.org/417/4012345000009type=https://ns.gs1.org/cbv/SDT-owning_partydestination=https://id.gs1.org/417/4012345000009type=https://ns.gs1.org/cbv/SDT-possessing_party');
+      const str5 = eventToPreHashedString({
+        errorDeclaration: {
+          declarationTime: '2020-01-15T00:00:00.000+01:00',
+          reason: 'incorrect_data',
+          'example:vendorExtension': {
+            '@xmlns:example': 'http://ns.example.com/epcis',
+            '#text': 'Test1',
+          },
+        },
+      }, {});
+      expect(str5).to.be.equal('errorDeclarationdeclarationTime=2020-01-14T23:00:00.000Zreason=https://ns.gs1.org/cbv/ER-incorrect_dataerrorDeclaration{http://ns.example.com/epcis}vendorExtension=Test1');
+    });
+  });
+
+  describe('global tests for pre-hashing', () => {
+    it('epcisDocHavingEventWithComment', () => {
+      for (let i = 0; i < epcisDocHavingEventWithComment.epcisBody.eventList.length; i += 1) {
```

Comment:
More succinct variable names here will help with readability, please. And clearer `it` clauses. You can also use clearer loop style with `forEach`:

```js
it('should pre-hash a document with a comment, () => {
  documentWithComment.epcisBody.eventList.forEach((event) => {
```